src/app/dashboard/photos/page.tsx

```
@@ -1,13 +1,274 @@
+          const msg = (() => { try { return JSON.parse(xhr.responseText)?.error?.message } catch { return undefined } })()
+          setUploads(prev => prev.map(u => u.id === uploadId ? { ...u, status: 'error', error: msg || `HTTP ${xhr.status}` } : u))
+        }
+        resolve()
+      }
+      xhr.onerror = () => {
+        setUploads(prev => prev.map(u => u.id === uploadId ? { ...u, status: 'error', error: 'Network error' } : u))
+        resolve()
+      }
+      setUploads(prev => prev.map(u => u.id === uploadId ? { ...u, status: 'uploading', progress: 0 } : u))
+      xhr.send(formData)
+    })
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)